File: backend/rating_scraper_api/CodeForces_api.py
# api.py

import requests

def fetch_codeforces_profile_api(handle):
    url = f"https://codeforces.com/api/user.info?handles={handle}"
    headers = {
        'User-Agent': 'Mozilla/5.0'
    }

    try:
        response = requests.get(url, headers=headers, timeout=10)
        response.raise_for_status()
        data = response.json()
        
        if data['status'] != 'OK':
            print(f"API error: {data.get('comment', 'Unknown error')}")
            return {'error': data.get('comment', 'Unknown error'), 'rating': 'N/A', 'handle': handle}
        
        user = data['result'][0]
        return {
            'rating': str(user.get('rating', 'N/A')),
            'handle': handle,
            'rank': user.get('rank', 'N/A'),
            'maxRating': str(user.get('maxRating', 'N/A'))
        }
    except requests.RequestException as e:
        print(f"Error fetching profile via API: {e}")
        return {'error': str(e), 'rating': 'N/A', 'handle': handle}
    except Exception as e:
        print(f"Unexpected error with Codeforces API: {e}")
        return {'error': str(e), 'rating': 'N/A', 'handle': handle}


def print_profile(profile):
    """Print the Codeforces rating"""
    if not profile:
        print("Unable to fetch Codeforces profile.")
    else:
        print(f"Codeforces Rating: {profile['rating']}")


if __name__ == "__main__":
    handle = input("Enter Codeforces handle: ")
    profile = fetch_codeforces_profile_api(handle)
    print_profile(profile)

File: backend-js/rating_scraper_api/codeforces_api.js

const axios = require('axios');

async function fetchCodeForcesProfile(handle) {
  const url = `https://codeforces.com/api/user.info?handles=${handle}`;
  const headers = {
    'User-Agent': 'Mozilla/5.0'
  };

  try {
    const response = await axios.get(url, { headers, timeout: 10000 });
    const data = response.data;
    
    if (data.status !== 'OK') {
      console.log(`API error: ${data.comment || 'Unknown error'}`);
      return { error: data.comment || 'Unknown error', rating: 'N/A', handle };
    }
    
    const user = data.result[0];
    return {
      rating: user.rating ? user.rating.toString() : 'N/A',
      handle,
      rank: user.rank || 'N/A',
      maxRating: user.maxRating ? user.maxRating.toString() : 'N/A'
    };
  } catch (error) {
    console.error(`Error fetching profile via API: ${error.message}`);
    return { error: error.message, rating: 'N/A', handle };
  }
}

function printProfile(profile) {
  if (!profile) {
    console.log("Unable to fetch Codeforces profile.");
  } else {
    console.log(`Codeforces Rating: ${profile.rating}`);
  }
}

module.exports = { fetchCodeForcesProfile, printProfile };
